### PR TITLE
Fixes #71 - remove dead, unused and unreachable code

### LINKED ISSUE #71: remove dead or unreachable code from repo
remove dead or unreachable code from repo

### PR DESCRIPTION
**Pull request by PaulCharlton**
Fixes #71 - remove dead, unused and unreachable code

### CODE REVIEW

File: pkg/util/loop/loop.go
Comment (PaulCharlton) [inline pkg/util/loop/loop.go:35]: will remove

File: e2e/internal/e2e/apptainercmd.go
Comment (ikaneshiro) [inline e2e/internal/e2e/apptainercmd.go:164]: We should leave all of these unused functions in `e2e/internal/` because they are part of our `e2e` test toolkit and may be valuable in the future

File: internal/pkg/cgroups/config_linux.go
Comment (ikaneshiro) [inline internal/pkg/cgroups/config_linux.go:196]: We should leave this for completeness of the `Get/Set` style api for this struct. This is actually used by this example plugin: https://github.com/apptainer/apptainer/blob/main/examples/plugins/config-plugin/main_linux.go#L54

File: pkg/image/driver.go
Comment (ikaneshiro) [inline pkg/image/driver.go:70]: This is important for this package API to allow plugins to register different types of image drivers: https://github.com/apptainer/apptainer/blob/main/examples/plugins/ubuntu-userns-overlay-plugin/main.go#L42

File: pkg/util/loop/loop.go
Comment (ikaneshiro) [inline pkg/util/loop/loop.go:34]: I dont see a reason to remove all of these constants. From my perspective they benefit the package API by having completeness in options available

**Review by ikaneshiro — CHANGES_REQUESTED** [review]

File: internal/pkg/cgroups/config_linux.go
Comment (PaulCharlton) [reply internal/pkg/cgroups/config_linux.go:196]: will preserve

File: e2e/internal/e2e/apptainercmd.go
Comment (PaulCharlton) [reply e2e/internal/e2e/apptainercmd.go:164]: I prefer not to keep things around simply because `someone might use it`

File: pkg/image/driver.go
Comment (PaulCharlton) [reply pkg/image/driver.go:70]: will preserve -- interesting that the dependency walker I use did not detect this.

File: pkg/util/loop/loop.go
Comment (PaulCharlton) [reply pkg/util/loop/loop.go:34]: having these constants in our project is a slippery slope at best.    can we find a more authoritative golang package, maintained by kernel developers, that provides these constants?

File: e2e/internal/e2e/apptainercmd.go
Comment (PaulCharlton) [reply e2e/internal/e2e/apptainercmd.go:164]: can we find a separate GOLANG e2e test framework and leave the maintenance to THAT community ?

Comment (ikaneshiro) [reply e2e/internal/e2e/apptainercmd.go:164]: all of our e2e framework is made ourselves to make sure we can run tests and inspect containers effectively. I am not aware of an e2e test framework in golang that will allow us to do what we are currently doing in our e2e tests and dont think it is worth migrating to a different framework to avoid mainlining something that has been effective and stable to date

File: pkg/util/loop/loop.go
Comment (ikaneshiro) [reply pkg/util/loop/loop.go:34]: im not aware of any loop package implementation in golang maintained by kernel developers, google or oci container maintainers would be our best hope, but they do not use loop devices in their container lifecycle. We were on our own when developing a lot of this code base because the operations against the OS is not very common in the go community

**Comment by DrDaveD** [comment]
So @PaulCharlton can you please restore the pieces that Ian requested?

File: e2e/internal/e2e/apptainercmd.go
Comment (PaulCharlton) [reply e2e/internal/e2e/apptainercmd.go:164]: reverting

**Comment by DrDaveD** [comment]
I believe that some of this may not be dead code after #85 is implemented so I think it would be better to hold this PR for that.

File: pkg/util/loop/loop.go
Comment (PaulCharlton) [reply pkg/util/loop/loop.go]: restoring

**Pull request closed without merging**
